### PR TITLE
Add a pre-commit hook to check flake8 compliance and prevent commits to main branch

### LINKED ISSUE #6: create pre-commit hooks
* hook for flake8
* prevent direct commits to main branch

### PR DESCRIPTION
**Pull request by 19201080**
Branches will need to be named either `feature/name_of_branch` or `fix/name_of_branch`.
Resolves #6.

### CODE REVIEW

**Review by jomaves — APPROVED** [review]

**Comment by oxullo** [comment]
@19201080 awesome! I'd suggest to add an entry to the README to describe how to install the reqs, install the hook and run a first manual test, so the process is clear.

File: setup.py [hunk 1]
```diff
@@ -10,7 +10,9 @@ def read(filename):
     filename = os.path.join(os.path.dirname(__file__), filename)
     text_type = type(u'')
     with io.open(filename, mode='r', encoding='utf-8') as fd:
-        return re.sub(text_type(r':[a-z]+:`~?(.*?)`'), text_type(r'``\1``'), fd.read())
```
Comment (oxullo) [inline setup.py:13]: @19201080 one small thing: this changeset is not relevant to the PR, try always to be minimal and pertinent, so it's easy to track the intent of your contribution

Comment (19201080) [reply setup.py:13]: Oups my bad! i'll know for next time

**Comment by oxullo** [comment]
@19201080 one last thing:

https://github.com/oxullo/sensorflux/pull/8/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R8

here I'd use

`$ python -m venv .venv`

to create a local virtualenv (which will be also ready to be picked up by pycharm). If you prefer to use pyenv, then instead of sourcing the activate you should:

`$ pyenv activate <ident>`

**Pull request merged**